Source/WebCore/page/Navigation.cpp

```diff
@@ -135,9 +135,20 @@ void Navigation::initializeForNewWindow(std::optional<NavigationNavigationType>
         if (shouldProcessPreviousNavigationEntries) {
             for (auto& entry : previousNavigation->m_entries)
                 m_entries.append(NavigationHistoryEntry::create(protectedScriptExecutionContext().get(), entry.get()));
-            m_currentEntryIndex = previousNavigation->m_currentEntryIndex;
-            if (navigationType == NavigationNavigationType::Reload) {
-                updateForActivation(currentItem.get(), navigationType);
+
+            // FIXME: This should handle Push, however somewhere before here the currentEntry of previousNavigation was updated
+            // to the new navigation so we get duplicate entries.
+            if (navigationType != NavigationNavigationType::Push) {
+                Ref previousEntry = m_entries[*previousNavigation->m_currentEntryIndex];
+
+                if (navigationType == NavigationNavigationType::Replace)
+                    m_entries[*previousNavigation->m_currentEntryIndex] = NavigationHistoryEntry::create(protectedScriptExecutionContext().get(), *currentItem);
+
+                m_currentEntryIndex = getEntryIndexOfHistoryItem(m_entries, *currentItem);
+
+                if (navigationType)
```

Comment:
I think I asked before, but when is this null? And do we have to test?

Reply:
Note, possibly is 204/205.

Reply:
Removed for now. Will see if any test hits this later.

Reply:
A handful of tests do fail because of this, will readd and see if it being unset in those cases makes sense.

Reply:
The failures were all forms/POST requests:

```
fast/history/form-submit-in-frame-via-onclick.html
fast/history/form-submit-in-frame.html
http/tests/navigation/post-frames.html
http/tests/navigation/postredirect-frames.html
```

I re-added this with a comment.